src/modules/Dashboard/features/Applet/Popups/DuplicatePopups/DuplicatePopups.tsx

```diff
@@ -190,36 +236,109 @@ export const DuplicatePopups = ({ onCloseCallback }: { onCloseCallback?: () => v
+        // but we don't have any proper workaround for this.
+        await setNewAppletReportServerPassword(
+          await publicEncrypt(
+            JSON.stringify({
+              password: encryptionDataRef.current.password ?? '',
+              privateKey: await getPrivateKey({
+                appletPassword: encryptionDataRef.current.password ?? '',
+                accountId: ownerId,
+              }),
+            }),
+            currentApplet?.reportPublicKey ?? '',
+          ),
+        );
+      }
+
+      handleDuplicateSuccess(newAppletSuccessResponse);
+    }
+
+    void effect();
+  }, [
+    currentApplet?.reportPublicKey,
+    currentAppletId,
+    duplicatePopupsVisible,
+    handleDuplicateSuccess,
+    includeReportServer,
+    newAppletSuccessResponse,
+    ownerId,
+    setAppletPrivateKey,
+    setNewAppletReportServerPassword,
+  ]);
+
+  useEffect(() => {
+    async function effect() {
+      // Fetch the report server properties of the applet
+      await dispatch(getApplet({ appletId: currentAppletId }));
+    }
+
+    void effect();
+  }, [currentAppletId, dispatch, getApplet]);
+
   return (
     <>
       <Modal
         open={nameModalVisible}
         onClose={nameModalClose}
         title={t('appletDuplication')}
-        onSubmit={setNameHandler}
+        onSubmit={onSubmitNameModal}
         buttonText={t('submit')}
         disabledSubmit={isLoading}
         data-testid={dataTestid}
       >
         <>
           {isLoading && <Spinner uiType={SpinnerUiType.Secondary} noBackground />}
           <StyledModalWrapper>
-            <form onSubmit={handleSubmit(setNameHandler)} noValidate>
-              <InputController
-                fullWidth
-                name="name"
-                control={control}
-                label={t('appletName')}
-                onChange={handleNameChange}
-                error={!!nameError}
-                data-testid={`${dataTestid}-name`}
-              />
+            <form onSubmit={handleSubmit(onSubmitNameModal)} noValidate>
+              <StyledFlexColumn sx={{ gap: 1.6 }}>
+                <Box>
+                  <InputController
+                    variant="outlined"
+                    fullWidth
+                    name="name"
+                    control={control}
+                    label={t('appletName')}
+                    onChange={handleNameChange}
+                    error={!!nameError}
+                    data-testid={`${dataTestid}-name`}
+                  />
+                  {nameError && (
+                    <StyledErrorText marginTop={0.5} marginBottom={0}>
+                      {nameError}
+                    </StyledErrorText>
+                  )}
+                </Box>
+                <Box>
+                  <CheckboxController
+                    name={'includeReportServer'}
+                    control={control}
+                    label={t('duplicateAppletReportServer')}
```

Comment:
`label` prop is expecting an `Element`
```suggestion
                   label={<StyledBodyLarge>{t('duplicateAppletReportServer')}</StyledBodyLarge>}
```